src/repositories/refArchitectureDao.js

```diff
@@ -24,6 +24,13 @@ refArchitectureSchema.statics = {
     delete: function (query) {
         return this.deleteOne(query);
     },
+
+    getAllProjectNames: function () {
+        return this.find(
+            { 'request_json.projectName': { $exists: true } },
+            { 'request_json.projectName': 1, _id: 0 }
+        ).then(results => results.map(result => result.request_json.projectName));
+    }
```

Comment:
A user can generate with same name as reference architecture.
If not allowed, they might be confused why the name is not allowed even though no other project exists with same name.